resource.go

```diff
@@ -92,6 +99,20 @@ func New(config Config) (*Resource, error) {
 		nodeCountFunc:            config.NodeCountFunc,
 		restClient:               config.RESTClient,
 		versionBundleVersionFunc: config.VersionBundleVersionFunc,
+		watcher:                  config.Watcher,
+	}
+
+	{
+		r.logger.Log("level", "debug", "message", "registering collector")
+
+		err := prometheus.Register(prometheus.Collector(r))
+		if IsAlreadyRegisteredError(err) {
+			r.logger.Log("level", "debug", "message", "collector already registered")
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		r.logger.Log("level", "debug", "message", "registered collector")
```

Comment:
Just doing the registration here now. We have an issue in the operator board to discuss if and how we want to do this in a more convenient way. See https://github.com/giantswarm/giantswarm/issues/4030.

Reply:
Maybe `Boot` function?